incubator/hnc/internal/validators/hncconfig.go

```diff
@@ -141,6 +147,91 @@ func (c *HNCConfig) validateType(ctx context.Context, t api.TypeSynchronizationS
 	return allow("")
 }
 
+func (c *HNCConfig) checkForest(inst *api.HNCConfiguration) admission.Response {
+	c.Forest.Lock()
+	defer c.Forest.Unlock()
+
+	// Get types that are changed from other modes to "Propagate" mode.
+	gvks := c.getNewPropagateTypes(inst)
+
+	// Check if user-created objects would be overwritten by these mode changes.
+	for gvk, _ := range gvks {
+		conflicts := c.checkConflictsForGVK(gvk)
+		if len(conflicts) != 0 {
+			msg := fmt.Sprintf("Cannot update configuration because setting type %q to 'Propagate' mode would overwrite user-created object(s):\n", gvk)
+			msg += strings.Join(c.checkConflictsForGVK(gvk), "\n")
```

Comment:
Should this just be `conflicts` instead of calling the function again?

Reply:
Ah right.